src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -296,16 +292,25 @@ - (void)clearTimer
 // All our callback dispatch must happen on _chipWorkQueue
 - (void)dispatchConnectionError:(CHIP_ERROR)error
 {
-    if (self.onConnectionError != nil) {
-        self.onConnectionError(self.appState, error);
+    if (_onConnectionError != nullptr) {
+        _onConnectionError(_appState, error);
     }
+    [self clearConnectionCallbacks];
 }
 
 - (void)dispatchConnectionComplete:(CBPeripheral *)peripheral
 {
-    if (self.onConnectionComplete != nil) {
-        self.onConnectionComplete(self.appState, BleConnObjectFromCBPeripheral(peripheral));
+    if (_onConnectionComplete != nullptr) {
+        _onConnectionComplete(self.appState, BleConnObjectFromCBPeripheral(peripheral));
     }
+    [self clearConnectionCallbacks];
```

Comment:
Might be safer to put _onConnectionComplete and _appState into on-stack variables, then clearConnectionCallbacks, then null-check and make the call using the stack things.

Same in dispatchConnectionError.